src/encoded/static/components/matrix_reference_epigenome.js

```diff
@@ -647,43 +756,80 @@ class MatrixPresentation extends React.Component {
-                                    </ModalBody>
-                                </Modal>
+                                <ClickableThumbnail expandedThumbnail={this.state.expandedThumbnail} expandThumbnail={this.expandThumbnail} />
+                                {this.state.expandedThumbnail ?
+                                    <BodyMapModal expandedThumbnail={this.state.expandedThumbnail} expandThumbnail={this.expandThumbnail} context={context} />
+                                : null}
                             </React.Fragment>
-                        }
-                    </TabPanel>
-                </div>
-            </div>
+                        : null}
+                        <div className="matrix__presentation">
+                            <div className={`matrix__label matrix__label--horz${!scrolledRight ? ' horz-scroll' : ''}`}>
+                                <span>{context.matrix.x.label}</span>
+                                {svgIcon('largeArrow')}
+                            </div>
+                            <div className="matrix__presentation-content">
+                                <div className="matrix__label matrix__label--vert"><div>{svgIcon('largeArrow')}{context.matrix.y.label}</div></div>
+                                {!this.state.organismChooserVisible ?
+                                    <div className="matrix__data" onScroll={this.handleOnScroll} ref={(element) => { this.scrollElement = element; }}>
+                                        {matrixConfig ?
+                                            <DataTable tableData={matrixConfig} />
+                                        : null}
+                                    </div>
+                                :
+                                    <React.Fragment>
+                                        <div className="matrix__data--empty" />
+                                        <Modal>
+                                            <ModalHeader closeModal={false} addCss="matrix__modal-header">
+                                                <h2>Reference epigenome &mdash; choose organism</h2>
+                                            </ModalHeader>
+                                            <ModalBody addCss="matrix-reference-epigenome__organism-selector">
+                                                <div>Organism to view in matrix:</div>
+                                                <div className="selectors">
+                                                    {Object.keys(organismTabs).map(organism => (
+                                                        // Encode the organism name into the <a> class for BDD testing.
+                                                        <a key={organism} className={`btn btn-info btn__selector--${organism.replace(/ /g, '-')}`} href={`${context['@id']}&replicates.library.biosample.donor.organism.scientific_name=${encoding.encodedURIComponent(organism)}`}>{organism}</a>
```

Comment:
The keys of organismTerms have a space. Is it a good idea to have keys DOM property that contain space?

Reply:
I think this will work a lot better as an array.